src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -844,11 +853,134 @@ private static string MakeShortCategoryString(string fullCategoryName)
+
+            // If it's not the last member, we can move further.
+            if (SelectedMemberIndex != selectedMemberGroup.Members.Count() - 1)
+            {
+                UnSelectOldMember();
+                SelectedMemberIndex++;
+                return;
+            }
+
+            // If it's last member, we should jump to next member group.
+            if (selectedMemberGroupIndex != selectedCategory.MemberGroups.Count() - 1)
+            {
+                UnSelectOldMember();
+                selectedMemberGroupIndex++;
+                SelectedMemberIndex = 0;
+                return;
+            }
+
+            // If it's last member group, we should jump to next category.
+            if (selectedCategoryIndex != SearchRootCategories.Count() - 1)
+            {
+                UnSelectOldMember();
+                selectedCategoryIndex++;
+                selectedMemberGroupIndex = 0;
+                SelectedMemberIndex = 0;
+                return;
+            }
+        }
+
+        /// <summary>
+        /// Used during library search key navigation. Counts previous selected member index.
+        /// Enables moving from bottom to top.
+        /// </summary>
+        public void SelectPrevious()
+        {
+            if (!SearchRootCategories.Any())
+                return;
 
+            var selectedCategory = SearchRootCategories[selectedCategoryIndex];
+            var selectedMemberGroup = selectedCategory.MemberGroups.ElementAt(selectedMemberGroupIndex);
+
+            // If it's not the first member, we can move back.
+            if (SelectedMemberIndex != 0)
+            {
+                UnSelectOldMember();
+                SelectedMemberIndex--;
+                return;
+            }
+
+            // If it's not the first member group, we can move back.
+            if (selectedMemberGroupIndex != 0)
+            {
+                UnSelectOldMember();
+                selectedMemberGroupIndex--;
+                // Select last member in member group. We are moving from bottom to top.
+                SelectedMemberIndex = selectedCategory.MemberGroups.ElementAt(selectedMemberGroupIndex).Members.Count() - 1;
+                return;
+            }
+
+            // If it's not the first category, we can move back.
+            if (selectedCategoryIndex != 0)
+            {
+                UnSelectOldMember();
+                selectedCategoryIndex--;
+                // Select last member group.
+                selectedMemberGroupIndex = SearchRootCategories[selectedCategoryIndex].MemberGroups.Count() - 1;
+                // Select last member in member group. We are moving from bottom to top.
+                SelectedMemberIndex = SearchRootCategories[selectedCategoryIndex].MemberGroups.
+                                            ElementAt(selectedMemberGroupIndex).Members.Count() - 1;
+                return;
+            }
+        }
```

Comment:
For the most part, the above two methods are identical, we shouldn't be having two code paths for similar functionality. Can we not do this:

```csharp
enum NavigationDirection
{
    Backward, Forward
}

public void MoveSelection(NavigationDirection direction)
{
    var offset = direction == NavigationDirection.Backward ? -1 : 1;

    selectedCategoryIndex += offset; // This will auto increment/decrement.
}
```

Does that make sense? :)